kong/runloop/wasm.lua

```diff
@@ -398,10 +398,18 @@ local function rebuild_state(db, version, old_state)
 
       for _, filter in ipairs(chain.filters) do
         if filter.enabled then
-          -- serialize all JSON configurations up front
-          if not filter.config and filter.json_config ~= nil then
-            filter.config = cjson_encode(filter.json_config)
-            filter.json_config = nil
+          -- Serialize all JSON configurations up front
+          --
+          -- NOTE: there is a subtle difference between a raw, non-JSON filter
```

Comment:
Reviewers: we've judged filters that require a JSON-encoded string to be an uncommon use case, compared to, say, an object or an array, which will still work perfectly fine.